src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -145,6 +146,13 @@
      */
     Optional<PluginContainer> getOwner();
 
+    /**
+     * Gets the default value this permission should have on this server
+     *
+     * @return The default value for this permission.
+     */
+    Tristate getDefaultValue();
```

Comment:
This contradicts the classes javadoc.
https://github.com/zml2008/SpongeAPI/blob/f0634898c257d89733f6a6d090f04c23efcac56a/src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java#L42

Also this might cause issues with the generic descriptions.
`Foo.<world>.doX`

For that reason we assigned only role strings with an actual value
https://docs.spongepowered.org/stable/en/plugin/permissions.html#permissiondescription

Reply:
> This contradicts the classes javadoc.

It doesn't really though - these aren't intended as _defaults_ that are automatically applied, but as _templates_, i.e. "here is the recommendation". They are still informational by the strictest sense of the word - as far as I understand it, they are not intended to be used as concrete groups, instead enabling a more global system akin to `nucleus setupperms`.

> Also this might cause issues with the generic descriptions. `Foo.<world>.doX`

Nothing says they have to be registered.

Reply:
> > This contradicts the classes javadoc.
> 
> It doesn't really though - these aren't intended as _defaults_ that are automatically applied, but as _templates_, i.e. "here is the recommendation". They are still informational by the strictest sense of the word - as far as I understand it, they are not intended to be used as concrete groups, instead enabling a more global system akin to `nucleus setupperms`.

If this only informational the an `assignRole("ALL", true)` would do the same and would also allow defining it for other roles. IIRC we even have some predefined role names somewhere.

Reply:
Thanks for pointing that out -- I'll change the class javadoc to match. Setting the default value in a description should update the default subject data.

Currently permissions descriptions are pretty useless -- and a good part of the reason why is that they don't really solve any problems that a list in a plugin's docs couldn't solve more easily.

@ST-DDT 's example permission is also something that's unlikely to make sense in the real world. I don't expect there to be many permissions that would incorporate a `world` parameter, since that's provided by contexts. I also expect most template parameters will be at the end of a permission string, not somewhere in the middle, so that wildcards are easier to apply -- meaning they can most likely automatically be replaced.

I'm going to add some `query` methods that take a Subject and check permissions based on the definition -- with support for replacing template parameters that are `String`s or `ContextKey`s. 

My end goal with permissions descriptions is to be able to replace the boilerplate in a class like https://gitlab.com/zml/featherchat/blob/master/src/main/java/ca/stellardrift/featherchat/Permissions.java#L49 or https://github.com/PEXPlugins/PermissionsEx/blob/master/permissionsex-fabric/src/main/kotlin/ca/stellardrift/permissionsex/fabric/hooks.kt#L91 -- with a list of `PermissionDescription` definitions.

---

As a side note regarding `/nucleus setupperms` -- that command should not exist, nucleus should provide its roles by registering `PermissionDescription`s, and a permissions plugin should be implementing the ability to copy data from a role template to a specific group.